tools/xc7frames2bit.cc

```diff
@@ -0,0 +1,398 @@
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <iterator>
+#include <string>
+#include <vector>
+
+#include <absl/strings/str_cat.h>
+#include <absl/strings/str_split.h>
+#include <absl/time/clock.h>
+#include <absl/time/time.h>
+#include <gflags/gflags.h>
+#include <prjxray/xilinx/xc7series/bitstream_writer.h>
+#include <prjxray/xilinx/xc7series/command.h>
+#include <prjxray/xilinx/xc7series/configuration.h>
+#include <prjxray/xilinx/xc7series/configuration_options_0_value.h>
+#include <prjxray/xilinx/xc7series/configuration_packet_with_payload.h>
+#include <prjxray/xilinx/xc7series/ecc.h>
+#include <prjxray/xilinx/xc7series/nop_packet.h>
+#include <prjxray/xilinx/xc7series/part.h>
+
+DEFINE_string(part_name, "", "");
+DEFINE_string(part_file, "", "Definition file for target 7-series part");
+DEFINE_string(
+    frm_file,
+    "",
+    "File containing a list of frame deltas to be applied to the base "
+    "bitstream.  Each line in the file is of the form: "
+    "<frame_address> <word1>,...,<word101>.");
+DEFINE_string(output_file, "", "Write bitsteam to file");
+
+namespace xc7series = prjxray::xilinx::xc7series;
+
+using PacketData = std::vector<uint32_t>;
+using BitHeader = std::vector<uint8_t>;
+using ConfigurationPackets =
+    std::vector<std::unique_ptr<xc7series::ConfigurationPacket>>;
+using FrameAddress2Frames =
+    std::map<xc7series::FrameAddress, std::vector<uint32_t>>;
+
+int readFrames(const std::string& frm_file_str, FrameAddress2Frames& frames) {
```

Comment:
Should readFrames be a function shared by xc7patch and xc7frames2bit?

Reply:
Yes, it could. I would just have to refactor the xc7patch to use it.